src/Api/HL/Controller/CoreController.php

```diff
@@ -421,6 +427,82 @@ public function getSession(Request $request): Response
         return new JSONResponse($session);
     }
 
+    #[Route(path: '/authorize', methods: ['GET', 'POST'], security_level: Route::SECURITY_NONE, tags: ['Session'])]
+    #[Doc\Route(
+        description: 'Authorize the API client using the authorization code grant type.',
+    )]
+    public function authorize(Request $request): Response
+    {
+        global $CFG_GLPI;
+        try {
+            $auth_request = Server::getAuthorizationServer()->validateAuthorizationRequest($request);
+            // Try loading session from Cookie
+            session_destroy();
+            ini_set('session.use_cookies', 1);
+            session_name("glpi_" . md5(realpath(GLPI_ROOT)));
+            @session_start();
+
+            $user_id = Session::getLoginUserID();
+            if ($user_id === false) {
+                // Redirect to login page
+                $scope = implode(',', $auth_request->getScopes());
+                $client_id = $auth_request->getClient()->getIdentifier();
+                $redirect_uri = $this->getAPIPathForRouteFunction(self::class, 'authorize');
+                $redirect_uri .= '?scope=' . $scope . '&client_id=' . $client_id . '&response_type=code&redirect_uri=' . urlencode($auth_request->getRedirectUri());
+                $redirect_uri = $CFG_GLPI['url_base'] . '/api.php/v2' . $redirect_uri;
+                return new Response(302, ['Location' => '/?redirect=' . rawurlencode($redirect_uri)]);
```

Comment:
Redirection should probably be done using an absolute URL. Indeed, GLPI can be located on a sub directory.